### PR TITLE
✅E2E Add file download testing in Docker

### PR DESCRIPTION
**Pull request by dominikkaminski**
**Changes:**

1. Add new BPMN download test 
2. Add GUI Server to desktop container to allow non-headless mode, which is needed for file downloading

PR: #1099 

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

**Comment by dominikkaminski** [comment]
@alexanderkasten I rebased to develop and added some more tests. 
Could you please rereview the pr?

**Comment by Paulomart** [comment]
@dominikkaminski Please rebase.

File: test/e2e/src/pages/exportDiagram.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export class ExportDiagram {
```
Comment (Paulomart) [inline test/e2e/src/pages/exportDiagram.ts]: I think this can be deleted.

File: test/jenkins_e2e.sh [hunk 2]
```diff
@@ -1,15 +1,110 @@
...
+    node node_modules/webdriver-manager update
+    node node_modules/webdriver-manager start &
 
-npm run jenkins-run-end-to-end-tests
+    # Start BPMN-Studio in background
+    npm start -- --port=9000 &
+
+    # Wait for required resources to be up and running.
+    echo "Waiting for SeleniumHub to start"
+    while ! curl --silent localhost:4444; do sleep 1; done
```
Comment (Paulomart) [inline test/jenkins_e2e.sh]:
```suggestion
    while ! curl --silent localhost:4444 > /dev/null; do sleep 1; done
```

**Comment by robpal94** [comment]
Any updates on this? @dominikkaminski @Paulomart @Vyperus

**Comment by Paulomart** [comment]
This is outdated. We moved to the [selenium dockerized](https://github.com/process-engine/selenium_dockerized) image and any changes to the `test/Dockerfile` must be submitted there.

**Comment by Vyperus** [comment]
Since the E2E Tests are getting refactored this will need some heavy refactoring as well.
I will close this PR for now, keeping the branch for later refactoring.

**Pull request closed without merging**